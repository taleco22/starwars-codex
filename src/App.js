import React from "react";
import {BrowserRouter, Route, Switch} from "react-router-dom";
import lightsaberMP3 from "./audio/sfx/lightsaber.mp3";
import lightsaberOGG from "./audio/sfx/lightsaber.ogg";
import mp3 from "./audio/starwars.mp3";
import ogg from "./audio/starwars.ogg";
import Categories from "./Routes/Categories";
import Home from "./Routes/Home";
import Subject from "./Routes/Subject";
import "./styles/css/index.css";
import Stars from "./styles/styled/Stars";
import lightning from "./video/lightning.mp4";

function App() {
  let isLoaded = false;
  window.onload = () => {
    isLoaded = true;
  };

  function prepareNavigation(isLoaded) {
    let actual;
    let btn;
    let intervalId;
    intervalId = setInterval(startNavigation, 200);

    function startNavigation(isLoaded) {
      if (isLoaded === false) return;
      clearInterval(intervalId);
      btn = [];
      actual = 0;
      btn = document.querySelectorAll("button");
      if (btn.length === 0) return prepareNavigation(isLoaded);
      document.addEventListener("keyup", applyFocus);
      btn.forEach((a) => a.addEventListener("pointerenter", applyFocus));
      btn.forEach((a) => a.addEventListener("pointerleave", applyFocus));
      btn[0].focus();
      applyFocus();
    }

    function applyFocus(e = {}) {
      if ((e.keyCode === 38) & (actual > 0)) {
        actual -= 1;
        btn[actual].focus();
      } else if ((e.keyCode === 40) & (actual < btn.length - 1)) {
        actual += 1;
        btn[actual].focus();
      } else if (e.target) {
        e.target.focus();
      }
      if (document.activeElement.nextSibling !== null) {
        btn.forEach((v) => {
          v.nextSibling && v.nextSibling.removeAttribute("data-openpreview");
        });
        document.activeElement.nextSibling.setAttribute("data-openpreview", true);
      }
    }
  }

  function start() {
    const audio = document.querySelector("#start");
    audio.load();
    audio.play();

    const video = document.querySelector("#video");
    video.classList.add("play");
    video.load();
    video.play();
    video.addEventListener("ended", () => video.classList.remove("play"));
  }

  return (
    <BrowserRouter>
      <audio autoPlay loop id="audio">
        <source id="ogg" src={ogg} type="audio/ogg" />
        <source id="mp3" src={mp3} type="audio/mpeg" />
      </audio>

      <audio id="start">
        <source id="ogg" src={lightsaberOGG} type="audio/ogg" />
        <source id="mp3" src={lightsaberMP3} type="audio/mpeg" />
      </audio>

      <video id="video">
        <source src={lightning} type='video/mp4; codecs="avc1.42E01E, mp4a.40.2"' />
      </video>

      <Stars />

      <Switch>
        <Route
          exact
          path="/"
          render={(props) => <Home {...props} start={start} fn={prepareNavigation(isLoaded)} />}
        />
        <Route
          exact
          path="/categories"
          render={(props) => <Categories {...props} fn={prepareNavigation(isLoaded)} />}
        />
        <Route
          exact
          path="/categories/characters"
          render={(props) => <Subject {...props} theme="people" fn={prepareNavigation(isLoaded)} />}
        />
        <Route
          exact
          path="/categories/starships"
          render={(props) => (
            <Subject {...props} theme="starships" fn={prepareNavigation(isLoaded)} />
          )}
        />
        <Route
          exact
          path="/categories/films"
          render={(props) => <Subject {...props} theme="films" fn={prepareNavigation(isLoaded)} />}
        />
        <Route
          exact
          path="/categories/vehicles"
          render={(props) => (
            <Subject {...props} theme="vehicles" fn={prepareNavigation(isLoaded)} />
          )}
        />
        <Route
          exact
          path="/categories/species"
          render={(props) => (
            <Subject {...props} theme="species" fn={prepareNavigation(isLoaded)} />
          )}
        />
        <Route
          exact
          path="/categories/planets"
          render={(props) => (
            <Subject {...props} theme="planets" fn={prepareNavigation(isLoaded)} />
          )}
        />
      </Switch>
    </BrowserRouter>
  );
}

export default App;
