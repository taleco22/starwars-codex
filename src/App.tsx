import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { audio, video } from "./assets";
import { Footer, Stars } from "./components/";
import store from "./pages/Subject/store";
import { Spinner } from "./pages/Subject/styled";

const Categories = React.lazy(() => import("./pages/Categories/Categories"));
const Home = React.lazy(() => import("./pages/Home/Home"));
const Subject = React.lazy(() => import("./pages/Subject/Subject"));

const App: React.FC = () => {
  React.useEffect(() => {
    const soundTrack: HTMLMediaElement | null = document.querySelector("#theme");
    soundTrack?.load();
    soundTrack?.play();
  }, []);

  return (
    <BrowserRouter>
      <audio loop={true} id="theme">
        <source id="ogg" src={audio.ogg} type="audio/ogg" />
        <source id="mp3" src={audio.mp3} type="audio/mpeg" />
      </audio>

      <audio id="lightsaberSound">
        <source id="ogg" src={audio.lightsaberOGG} type="audio/ogg" />
        <source id="mp3" src={audio.lightsaberMP3} type="audio/mpeg" />
      </audio>

      <video id="lightsaberVideo">
        <source src={video.lightning} type='video/mp4; codecs="avc1.42E01E, mp4a.40.2"' />
      </video>

      <Stars />

      <Switch>
        <React.Suspense fallback={<Spinner />}>
          <Route exact path="/" component={Home} />
          <Route exact path="/categories" component={Categories} />
          <Provider store={store}>
            <Route exact path="/categories/:theme" component={Subject} />
          </Provider>
        </React.Suspense>
      </Switch>

      <Footer />
    </BrowserRouter>
  );
};

export default App;
