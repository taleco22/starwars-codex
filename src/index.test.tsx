import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import App from "./App";

describe("index.tsx", () => {
  it("renders without crashing ", () => {
    const component = renderer.create(<App />).toJSON();

    expect(component).toMatchSnapshot();

    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
