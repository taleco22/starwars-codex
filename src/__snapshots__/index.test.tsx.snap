// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.tsx renders without crashing  1`] = `
Array [
  <audio
    id="theme"
    loop={true}
  >
    <source
      id="ogg"
      src="starwars.ogg"
      type="audio/ogg"
    />
    <source
      id="mp3"
      src="starwars.mp3"
      type="audio/mpeg"
    />
  </audio>,
  <audio
    id="lightsaberSound"
  >
    <source
      id="ogg"
      src="lightsaber.ogg"
      type="audio/ogg"
    />
    <source
      id="mp3"
      src="lightsaber.mp3"
      type="audio/mpeg"
    />
  </audio>,
  <video
    id="lightsaberVideo"
  >
    <source
      src="lightning.mp4"
      type="video/mp4; codecs=\\"avc1.42E01E, mp4a.40.2\\""
    />
  </video>,
  <div
    className="sc-AxhUy fTiWoS layer1"
  />,
  <div
    className="sc-AxhUy fTiWoS layer1 a"
  />,
  <div
    className="sc-AxhUy fTiWoS layer2"
  />,
  <div
    className="sc-AxhUy fTiWoS layer2 a"
  />,
  <div
    className="sc-AxhUy fTiWoS layer3"
  />,
  <div
    className="sc-AxhUy fTiWoS layer3 a"
  />,
  <main
    className="sc-AxgMl sc-fznyAO jowjTH"
  >
    <img
      alt="Star Wars"
      className="sc-fzqNJr kAiDBy"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "star-logo.svg",
        }
      }
    />
    <div
      className="sc-fzqBZW hymcIh"
    >
      <button
        className="sc-AxmLO sc-fzokOt fzGgUX"
        onClick={[Function]}
      >
        Start
      </button>
    </div>
  </main>,
  <div
    className="sc-AxjAm iWNzyN"
  >
    <div
      className="sc-AxirZ hatWLT"
    >
      <div
        className="sc-AxiKw imGYnc"
      >
        <div
          className="sc-AxhCb hWGxmf"
        >
          <img
            alt="up-arrow"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "up-arrow.svg",
              }
            }
          />
        </div>
        - Up
      </div>
      <div
        className="sc-AxiKw imGYnc"
      >
        <div
          className="sc-AxhCb hWGxmf"
        >
          <img
            alt="down-arrow"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "down-arrow.svg",
              }
            }
          />
        </div>
        - Down
      </div>
      <div
        className="sc-AxiKw imGYnc"
      >
        <div
          className="sc-AxhCb hWGxmf"
        >
          <img
            alt="enter-key"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "enter.svg",
              }
            }
          />
        </div>
        - Confirm
      </div>
    </div>
  </div>,
]
`;
